KILTprotocol/local-parachain-setup
# feat: add parachain registration script

## fixes KILTProtocol/ticket#1470
* Adds script to automatically onboard the provided parachain 
* Increases lease to 365 days
* Speeds up onboarding time by pushing parachain to next session (instead of 2 sessions in the future)
  * NOTE: Unfortunately, it is not possible to trigger a new session via an extrinsic. For development purposes we should probably modify the Westend configuration slightly and build our own image. We just need to decrease the duration of a session.
* Adds environment variables for all used ports to reduce merge conflicts and simplify running this application for the whole team

The image for the node script should always be rebuilt because of the potential change in the `WASM` and `Genesis.hex` of the parachain which need to be copied into the scripts directory. AFAIK, there is no way to access parent directories when building the docker image which is why we need to copy both files into the scripts directory in the first place.

## How to test:
Run the script, wait ~30 seconds and then connect to the [Polkadot Apps Parachain Section](https://polkadot.js.org/apps/#/parachains/parathreads). You should see the Parathread 2000 to be onboarding for less than one hour if using the Peregrine Prod/Staging spec.
```bash
./run.sh
```

## Checklist:
